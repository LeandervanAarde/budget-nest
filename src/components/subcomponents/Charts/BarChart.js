import React from 'react';
import { Bar } from 'react-chartjs-2';
import { CanvasHTMLAttributes } from 'react';
import 'chart.js/auto';
const BarChart = (props) => {
    return (
        
     <Bar  data={{
        labels: props.name,
        datasets:[{
            data: props.data, 
            backgroundColor: [
                " #48D87F",
                "#2494FA",
                "#7F6EE1"
             ], 
        }],
     }}
     height={320}
     options={{
         responsive: true,
         maintainAspectRatio: true,
         plugins: {
             decimation: {
                 enabled: true,
             },
             title: {
                 display: true,
                 text: "HouseholdIncome",
                 position: "left",
                 color: "white",
                 font: {
                     size: 24,
                     weight: "bold",

                 }
             },
             legend: {
                 position: "bottom",
                 display: false,
                 labels: {
                     display: false,
                     padding: 30,
                     boxWidth: 30,
                     boxHeight: 20,
                 }
             }
         },
         indexAxis: 'x',
         
         animation: {
             easing: 'easeInOutBounce',
             duration: 1800,
             numSteps: 2,
         },
         datasets: {
             bar: {
                 base: 0,
                 borderWidth: 3,
                 hoverBorderColor: "white",
                 hoverBorderWidth: 3,
                 borderRadius: 10,
             }
         }
     }}
     />
    );
};

export default BarChart;