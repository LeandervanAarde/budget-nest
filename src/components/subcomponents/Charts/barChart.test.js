import { render, screen, } from "@testing-library/react";
import { ReactTestRenderer , create, act} from "react-test-renderer";
import BarChart from "./BarChart";
import renderer from 'react-test-renderer';
const TestRenderer = require('react-test-renderer');


it('render correctly..', () =>{
    const testRender = TestRenderer.create(
        <BarChart name={["Leander", "Joshua", "Dae", "Matthew", "Camerom"]} data={[12,23,676,211,4300]}/>
        );
    const testInstance = testRender.root;
    expect(testInstance.findByType(BarChart).props.name).toBe["Leander", "Joshua", "Dae", "Matthew", "Camerom"];
});

it("matches snap", async() =>{
    let root;
    await act(() =>{
         root = create( <BarChart name={["Leander", "Joshua", "Dae", "Matthew", "Camerom"]} data={[12,23,676,211,4300]}/>);
    });

    expect(root.toJSON()).toMatchSnapshot();

    act(() =>{
        root.update(<BarChart name={["Leander", "Joshua", "Dae", "Matthew", "Camerom", "Brosamino"]} data={[12,23,676,211,4300,5000]}/>);
        expect(root.toJSON()).toMatchSnapshot();
    })

    console.log(root.toTree)
})


