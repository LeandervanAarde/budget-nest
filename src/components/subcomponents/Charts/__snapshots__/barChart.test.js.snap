// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snap 1`] = `null`;

exports[`matches snap 2`] = `null`;
